lxd/containers_post.go

```diff
@@ -14,12 +14,57 @@ import (
 	"github.com/lxc/lxd/shared"
 )
 
-func btrfsCopyImage(hash string, name string, d *Daemon) error {
-	dpath := shared.VarPath("lxc", name)
-	imagefile := shared.VarPath("images", hash)
-	subvol := fmt.Sprintf("%s.btrfs", imagefile)
+func btrfsCmdIsInstalled() bool {
+	/*
+	 * Returns true if the "btrfs" tool is in PATH else false.
+	 *
+	 * TODO: Move this to the main code somewhere and call it once?
+	 */
+	out, err := exec.Command("which", "btrfs").Output()
+	if err != nil || len(out) == 0 {
+		return false
+	}
+
+	return true
+}
 
-	return exec.Command("btrfs", "subvolume", "snapshot", subvol, dpath).Run()
+func btrfsIsSubvolume(subvolPath string) bool {
+	/*
+	 * Returns true if the given Path is a btrfs subvolume else false.
+	 */
+	if !btrfsCmdIsInstalled() {
+		return false
+	}
+
+	out, err := exec.Command("btrfs", "subvolume", "show", subvolPath).CombinedOutput()
+	if err != nil || strings.HasPrefix(string(out), "ERROR: ") {
```

Comment:
I've tested that and `btrfs` always exits with `0`

##### NO Error:

    root@srv01:/home/pcdummy# btrfs subvolume show /; echo $?
    /
            Name:                   @root
            uuid:                   e4041a49-60fa-f24a-9e23-5d22c5a3c863
            Parent uuid:            -
            Creation time:          2015-02-27 23:41:37
            Object ID:              258
            Generation (Gen):       338616
            Gen at creation:        7
            Parent:                 5
            Top Level:              5
            Flags:                  -
            Snapshot(s):
    0

##### Error:

    root@srv01:/home/pcdummy# btrfs subvolume show /tmp; echo $?
    ERROR: '/tmp' is not a subvolume
    0

Thanks a lot for your review, @tych0